src/Spice86.Core/Emulator/InterruptHandlers/Dos/Xms/ExtendedMemoryManager.cs

```diff
@@ -0,0 +1,490 @@
+﻿namespace Spice86.Core.Emulator.InterruptHandlers.Dos.Xms;
+
+using Serilog.Events;
+
+using Spice86.Core.Emulator.CPU;
+using Spice86.Core.Emulator.Extensions;
+using Spice86.Core.Emulator.InterruptHandlers;
+using Spice86.Core.Emulator.InterruptHandlers.Common.MemoryWriter;
+using Spice86.Core.Emulator.InterruptHandlers.Common.RoutineInstall;
+using Spice86.Core.Emulator.Memory;
+using Spice86.Shared.Emulator.Memory;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+/// <summary>
+/// Provides DOS applications with XMS memory.
+/// </summary>
+public sealed class ExtendedMemoryManager : InterruptHandler, IMemoryDevice {
+    private int _a20EnableCount;
+    private readonly LinkedList<XmsBlock> _xmsBlocksLinkedList = new();
+    private readonly SortedList<int, int> _xmsHandles = new();
+    
+    public const ushort CallbackAsmFunctionSegment = 0xD000;
+
+    /// <summary>
+    /// The size of available XMS Memory, in bytes.
+    /// </summary>
+    public const uint XmsMemorySize = 8 * 1024 * 1024;
+    
+    private class XmsControlFunctionWriter : IAssemblyRoutineWriter {
+        public SegmentedAddress WriteAssemblyInRam(MemoryAsmWriter memoryAsmWriter) {
+            SegmentedAddress startAddress = new(CallbackAsmFunctionSegment, 0);
+            memoryAsmWriter.CurrentAddress = startAddress;
+            memoryAsmWriter.WriteJmpNear(0x3);
+            memoryAsmWriter.WriteNop();
+            memoryAsmWriter.WriteNop();
+            memoryAsmWriter.WriteNop();
+            return startAddress;
+        }
+    }
+
+    public ExtendedMemoryManager(IMemory memory, Cpu cpu, AssemblyRoutineInstaller assemblyRoutineInstaller, ILoggerService loggerService) : base(memory, cpu, loggerService) {
+        assemblyRoutineInstaller.InstallAssemblyRoutine(new XmsControlFunctionWriter(), "XMS Control Function");
+        _memory.RegisterMapping(XmsBaseAddress, XmsMemorySize, this);
+        _xmsBlocksLinkedList.AddFirst(new XmsBlock(0, 0, XmsMemorySize, false, memory, XmsBaseAddress));
+        FillDispatchTable();
+    }
+
+    /// <summary>
+    /// Specifies the starting physical address of XMS.
+    /// </summary>
+    public const uint XmsBaseAddress = 0x10FFF0;
+
+    /// <summary>
+    /// Total number of handles available at once.
+    /// </summary>
+    private const int MaxHandles = 128;
+
+    /// <summary>
+    /// Gets the largest free block of memory in bytes.
+    /// </summary>
+    public uint LargestFreeBlock => GetFreeBlocks().FirstOrDefault()?.Length ?? 0;
+    
+    /// <summary>
+    /// Gets the total amount of free memory in bytes.
+    /// </summary>
+    public long TotalFreeMemory => GetFreeBlocks().Sum(b => b.Length);
+    
+    /// <summary>
+    /// Gets the total amount of extended memory.
+    /// </summary>
+    public int ExtendedMemorySize => (int)XmsMemorySize;
+
+    public override byte VectorNumber => 0x43;
```

Comment:
Are you sure this is the correct vector number? http://www.ctyme.com/intr/int-43.htm -> seems to be something else

Reply:
Dune only uses the BIOS function Copy Extended Memory (0x87)